monri/src/main/java/com/monri/android/activity/ConfirmPaymentActivity.java

```diff
@@ -51,7 +74,7 @@ protected void onCreate(Bundle savedInstanceState) {
                 .set("meta.library", MonriUtil.library(getApplicationContext()))
                 .set("meta.library_version", BuildConfig.MONRI_SDK_VERSION);
 
-        monri = new Monri(this, apiOptions);
+        monri = new Monri(((ActivityResultCaller) this), apiOptions);
```

Comment:
@adnanomerovic is this safe to do? Is there a way for this to fail with class cast exception?

Reply:
Runtime ClassCastException will be thrown.